File: Students/Nirali Ganvit/notes-backend/src/main/java/com/example/notes_backend/service/NoteService.java
package com.example.notes_backend.service;

import com.example.notes_backend.model.Note;
import java.util.List;

public interface NoteService {

    /**
     * Creates a new note and persists it in the database.
     *
     * @param note the note to be created
     * @return the created note with generated id and timestamps
     */
    Note createNote(Note note);

    /**
     * Updates an existing note identified by its ID.
     *
     * @param id the unique identifier of the note to update
     * @param note the note containing updated fields
     * @return the updated note
     */
    Note updateNote(String id, Note note);

    /**
     * Deletes a note from the database by its ID.
     *
     * @param id the unique identifier of the note to delete
     */
    void deleteNote(String id);

    /**
     * Retrieves all notes from the database.
     *
     * @return a list of all notes
     */
    List<Note> getAllNotes();

    /**
     * Retrieves a single note by its ID.
     *
     * @param id the unique identifier of the note
     * @return the note if found
     */
    Note getNoteById(String id);

    /**
     * Toggles the favorite status of a note.
     *
     * @param id the unique identifier of the note
     * @return the updated note
     */
    Note favoriteNote(String id);

    /**
     * Toggles the archived status of a note.
     *
     * @param id the unique identifier of the note
     * @return the updated note
     */
    Note archiveNote(String id);

    /**
     * Retrieves all notes marked as favorites.
     *
     * @return a list of favorite notes
     */
    List<Note> getFavoriteNotes();

    /**
     * Retrieves all notes marked as archived.
     *
     * @return a list of archived notes
     */
    List<Note> getArchivedNotes();
}

File: Students/Dhruv Chaudhari/notes-frontend/src/services/NoteService.js
import axios from "axios";

const API = `${import.meta.env.VITE_API_BASE_URL}/notes`;

const noteService = {

    getAllNotes: async () => {
        const response = await axios.get(API);
        return response.data;
    },

    getNoteById: async (id) => {
        const response = await axios.get(`${API}/${id}`);
        return response.data;
    },

    createNote: async (note) => {
        const response = await axios.post(API, note);
        return response.data;
    },

    updateNote: async (id, noteDetails) => {
        const response = await axios.put(`${API}/${id}`, noteDetails);
        return response.data;
    },

    deleteNote: async (id) => {
        const response = await axios.delete(`${API}/${id}`);
        return response.data;
    },

    favoriteNote: async (id) => {
        const response = await axios.put(`${API}/${id}/favorite`);
        return response.data;
    },

    archiveNote: async (id) => {
        const response = await axios.put(`${API}/${id}/archive`);
        return response.data;
    },

    getFavoriteNotes: async () => {
        const response = await axios.get(`${API}/favorites`);
        return response.data;
    },

    getArchivedNotes: async () => {
        const response = await axios.get(`${API}/archive`);
        return response.data;
    }

};

export default noteService;